src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -248,11 +248,15 @@ void GcsServer::InitGcsHeartbeatManager(const GcsInitData &gcs_init_data) {
 }
 
 void GcsServer::InitGcsResourceManager(const GcsInitData &gcs_init_data) {
-  RAY_CHECK(gcs_table_storage_ && cluster_resource_scheduler_);
+  RAY_CHECK(gcs_table_storage_ && cluster_resource_scheduler_ && cluster_task_manager_);
   gcs_resource_manager_ = std::make_shared<GcsResourceManager>(
       gcs_table_storage_,
       cluster_resource_scheduler_->GetClusterResourceManager(),
-      scheduling::NodeID(local_node_id_.Binary()));
+      scheduling::NodeID(local_node_id_.Binary()),
+      /*get_gcs_node_resource_usage=*/
+      [this](rpc::ResourcesData &data) {
+        cluster_task_manager_->FillPendingActorInfo(data);
```

Comment:
why not just put cluster_task_manager_ into GcsResourceManager like cluster_resource_scheduler?

Reply:
It was meant to reduce the direct dependency on other classes, because `cluster_task_manager_` is only used in one place inside `gcs_resource_manager`. But actually I'm not sure which way is more light-weight (or conventional). Ideas? @scv119 @jjyao @wumuzi520

Reply:
I feel if we depend on other classes, we just do it **explicitly** so that from the member of the class we'll know this. 
If passing by callback, it's actually hard to figure out the dependence graphs.

IMO, the callback is to be used when API is not supposed to know the provider of the caller. But here cluster_task_manager_/gcs_resource_manager seems not this case.